shared/teams/add-people/container.js

```diff
@@ -1,39 +1,47 @@
 // @flow
-import {connect} from 'react-redux'
 import * as Creators from '../../actions/teams/creators'
 import * as SearchCreators from '../../actions/search/creators'
 import AddPeople from '.'
 import {HeaderHoc} from '../../common-adapters'
 import {navigateAppend} from '../../actions/route-tree'
-import {compose, withPropsOnChange} from 'recompose'
-
-import type {TypedState} from '../../constants/reducer'
+import {
+  connect,
+  compose,
+  withHandlers,
+  withPropsOnChange,
+  withState,
+  type TypedState,
+} from '../../util/container'
 
 const mapStateToProps = (state: TypedState, {routeProps}) => ({
   name: routeProps.get('teamname'),
 })
 
 const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, routeProps}) => ({
-  onClose: () => dispatch(navigateUp()),
   onAddPeople: (role: string) => {
     dispatch(Creators.addPeopleToTeam(routeProps.get('teamname'), role))
     dispatch(navigateUp())
     dispatch(Creators.getTeams())
     dispatch(SearchCreators.clearSearchResults('addToTeamSearch'))
     dispatch(SearchCreators.setUserInputItems('addToTeamSearch', []))
   },
+  onClose: () => dispatch(navigateUp()),
   onOpenRolePicker: (role: string, onComplete: string => void) => {
-    dispatch(navigateAppend([{selected: 'controlledRolePicker', props: {selectedRole: role, onComplete}}]))
+    dispatch(navigateAppend([{props: {onComplete, selectedRole: role}, selected: 'controlledRolePicker'}]))
```

Comment:
`props: {..., allowOwner: false}` to alleviate black bar `cannot invite non-keybase users as owners` error